incubator/hnc/pkg/testutils/testutils.go

```diff
@@ -229,7 +232,49 @@ func RunCommand(cmdln ...string) (string, error) {
 	stdout, err := cmd.CombinedOutput()
 	return string(stdout), err
 }
+// CreateNamespace creates the specified namespace with canned testing labels making it easier
+// to look up and delete later.
+func CreateNamespace( ns string)   {
+	CreateNamespaceWithLabels(ns,"", testingNamespaceLabels)
+}
+// CreateSubnamespace creates the specified namespace in the parent namespace with canned testing labels making it easier
+// to look up and delete later.
+func CreateSubnamespace( ns string,parent string)   {
+	CreateNamespaceWithLabels(ns,parent, testingNamespaceLabels)
+}
+
+// CreateNamespaceWithLabels creates a namespace or a subnamespace (if a parent is specified) and marks the namespace
```

Comment:
Ok , sure. yes, currently only CreateNamespace  and CreateSubnamespace  use it but I figured if there might be future test cases where one might want to pass a list of labels and delete by those by labels it might be useful to leave these there .If you think its not needed , i will remove and simplify it.

Reply:
I wouldn't worry about future test cases for now - it's easy to refactor stuff out later if we need it, but until then, it will just confuse anyone looking at this code. I always go for "simplicity now" unless there's a compelling reason otherwise, and I don't see one in this case.

Thanks!

Reply:
Sounds good. 
Thanks

Reply:
I have made the relevant changes . Please check .

Reply:
This seems like a lot of complexity, and I _think_ it's just driven by the needs of `RecoverHNC`, right? I think I'd prefer that `CreateNamespace` and `CreateSubnamespace` should just be two-line functions, and let `RecoverHNC` do everything manually (since it doesn't even need the labels in the first place).

Reply:
The whole timeout parameter is driven by the RecoverHNC yes. I had it like you suggested before without touching RecoverHNC but then figured for consistency to use this common function. But yes, i like the idea of reverting RecoverHNC to the manual way and removing all this timeout complexity. I added the CreateNamespaceWithLabels common function since i didn't like repeating the line 
	MustRun("kubectl label --overwrite ns", ns,testingNamespaceLabel+"=true")
in both the functions.

Reply:
I understand wanting to avoid repetition, but in this case I think having two "regular" functions call a common implementation makes the logic harder to understand. For one line, I think copy and paste is fine - at most I could imagine a utility function that you call like `labelNS(ns, testingNamespaceLabel, "true")` (eg factor out the common part, as opposed to wrapping a superset of functionality in a simpler interface) but personally I wouldn't bother with that. You can if you want though.